### PR TITLE
docs: Add English translations

### PR DESCRIPTION
**Pull request by tadatuta**

### CODE REVIEW

File: docs/en/3-api.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
+# API
+
+* [ Choosing an engine, compiling and applying templates] (#choose-engine)
+* [Adding templates] (#add-templates) 
+* [Settings] ( #settings) 
+  - [ Delimiters in names of BEM entities ] (#delimeters)
+  - [ Extending BEMContext] (#bemcontext) 
+* [ Bundling] ( #bundle) 
+
+## <a name="choose-engine"> </a>Choosing an engine, compiling and applying templates
```
Comment (miripiruni) [inline docs/en/3-api.md]: `<a name="choose-engine"></a>`

File: docs/en/3-api.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
+# API
+
+* [ Choosing an engine, compiling and applying templates] (#choose-engine)
+* [Adding templates] (#add-templates) 
+* [Settings] ( #settings)
```
Comment (miripiruni) [inline docs/en/3-api.md]: There is more settings in russian docs. Should we add it later?

File: docs/en/3-api.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
+# API
+
+* [ Choosing an engine, compiling and applying templates] (#choose-engine)
+* [Adding templates] (#add-templates) 
+* [Settings] ( #settings) 
+  - [ Delimiters in names of BEM entities ] (#delimeters)
```
Comment (miripiruni) [inline docs/en/3-api.md]: useless spaces

File: docs/en/3-api.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+var bemxjst = require('bem-xjst');
+var bemhtml = bemxjst.bemhtml;
+
+// Add a template
+var templates = bemhtml.compile(function() {
+    block('quote').tag()('q');
+});
+
+// Add data
+var bemjson = { block: 'quote', content: 'Пришел, увидел, отшаблонизировал.' };
```
Comment (miripiruni) [inline docs/en/3-api.md]: `Veni, vidi, templati` :)

File: docs/en/3-api.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+});
+
+html = templates.apply(bemjson);
+// Now the HTML tag is h2 instead of h1:
+// html: '<h2 class="header">Documentation</h2>'
+```
+
+If you need to [bundle] (https://en.bem.info/method/build/#build-results) all the templates, the most efficient way is to use the [generate] (#generate) method.
+
+## <a name="settings"> </a> Settings
```
Comment (miripiruni) [inline docs/en/3-api.md]: space

File: docs/en/3-api.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        elem: 'head',
+        elemMods: { type: 'short' }
+    }
+};
+
+var html = templates.apply(bemjson);
+```
+The resulting `html` contains the string:
+```html
+<div class="page page_theme_gray"> <div class="page__head page__head_type_short"> </div> </div>
```
Comment (miripiruni) [inline docs/en/3-api.md]: Why spaces?

File: docs/en/3-api.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+    }
+};
+
+var html = templates.apply(bemjson);
+```
+The resulting `html` contains the string:
+```html
+<div class="page page_theme_gray"> <div class="page__head page__head_type_short"> </div> </div> 
+```
+You can find more information about [naming conventions] ( https://en.bem.info/method/naming-convention/) on bem.info.
```
Comment (miripiruni) [inline docs/en/3-api.md]: spaces :)

File: docs/en/3-api.md [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+// Input data
+var bemjson = { block: 'b' };
+
+// Apply templates
+var html = templates.apply(bemjson);
+```
+
+The resulting `html` contains the string:
+```html
+<div class="b"> Hello, templates</div>
```
Comment (miripiruni) [inline docs/en/3-api.md]: `&gt;Hello`

File: docs/en/4-data.md [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+```js
+// BEMJSON
+{
+   block: 'link',
+   js: { title: 'film "Sid & Nancy"' }
+}
+```
+Result of BEMHTML templating:
+```html
+<div class="link i-bem" data-bem='{"link":{title:"&#39;Sid &amp; Nancy&#39;"}}'> </div>
```
Comment (miripiruni) [inline docs/en/4-data.md]: space

File: docs/en/4-data.md [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+
+### attrs
+
+`{Object}` hash of HTML attributes. Attribute values are [escaped using the attrEscape function](6-templates-context.md#attrescape). Example:
+```js
+// BEMJSON
+{
+    attrs: {
+        id: 'anchor',
+        name: 'Мультфильм "Tom & Jerry"'
```
Comment (miripiruni) [inline docs/en/4-data.md]: Cartoon

File: docs/en/4-data.md [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+### cls
+
+`{String}` HTML class or classes (separated by spaces) that do not belong to the BEM subject domain (block-element-modifier). For example, the use of [microformats](http://microformats.org/) or semantic markup from [schema.org](https://schema.org/).
+
+```js
+// BEMJSON
+{
+    block: 'user',
+    cls: 'h-card p-name',
+    content: 'Андрей Линде'
```
Comment (miripiruni) [inline docs/en/4-data.md]: `Andrei Linde`

File: docs/en/5-templates-syntax.md [hunk 12]
```diff
@@ -0,0 +1,352 @@
...
+```
+Both templates are applied. Result:
+```html
+<body class="page__content page__content_type_index mixed"></body>
+```
+
+
+### match
+```js
+/** * @param {Function} Checking a custom condition. The result is converted to `Boolean`. */ match(function() { return … })
```
Comment (miripiruni) [inline docs/en/5-templates-syntax.md]: Multiple lines

File: docs/en/5-templates-syntax.md [hunk 13]
```diff
@@ -0,0 +1,352 @@
...
+
+```js
+block('link').elem('icon')
+block('link').elem('text')
+```
+
+...you can use a nested structure. The subpredicates are placed in the body of a shared subpredicate and separated with commas.
+```js
+block('link')(
+    elem('icon')(тело_шаблона_1),
```
Comment (miripiruni) [inline docs/en/5-templates-syntax.md]: template_body_1

File: docs/en/5-templates-syntax.md [hunk 14]
```diff
@@ -0,0 +1,352 @@
...
+There is no restriction on the nesting level of subpredicates.
+
+
+## <a name="body"></a>Template body
+The body of a template represents instructions for generating a templating result for the current BEMJSON node.
+
+The process of templating each node of the input data consists of phases called modes.. Each mode is responsible for generating a separate part of the result.. For example, for BEMHTML this might be an HTML tag, HTML class, HTML attributes, tag contents, and so on.
+
+```html
+<!-- `def` mode--> <div <!-- the opening and closing element of an HTML tag depends on the `tag` mode--> class=" link <!-- the name of the BEM entity is formed from the fields block, mods, elem, elemMods --> mixed <!-- `mix` mode --> cls <!-- `cls` mode--> i-bem <!-- `js` mode --> " <!-- `bem` mode --> data-bem='{"link":{}}' <!-- `attr` mode --> id="my-dom-node" > Tag content <!-- `content` mode --> </div> <!-- the closing element of the tag also depends on the `tag` mode -->
```
Comment (miripiruni) [inline docs/en/5-templates-syntax.md]: Formatting

File: docs/en/5-templates-syntax.md [hunk 15]
```diff
@@ -0,0 +1,352 @@
...
+block('b').content('test');
+// This will throw the BEMHTML error: Predicate should not have arguments.
+
+// Correct:
+block('b').content()('test');
+```
+
+For input data:
+```js
+{ block: 'link', url: 'https://yandex.ru', content: 'Яндекс' }
```
Comment (miripiruni) [inline docs/en/5-templates-syntax.md]: Yandex

File: docs/en/5-templates-syntax.md [hunk 16]
```diff
@@ -0,0 +1,352 @@
...
+block('quote').wrap()(function() {
+    return {
+        block: 'wrap',
+        content: this.ctx
+    };
+});
+```
+Result of BEMHTML templating:
+```html
+<div class="wrap"> <div class="quote">Docendo discimus</div> </div>
```
Comment (miripiruni) [inline docs/en/5-templates-syntax.md]: Why spaces?

File: docs/en/5-templates-syntax.md [hunk 17]
```diff
@@ -0,0 +1,352 @@
...
+        ];
+    }),
+    elem('input').tag()('input'),
+    elem('label').tag()('label')
+);
+```
+
+Result of BEMHTML templating:
+```html
+<div class="control"> <label class="control__label" for="username-control"></label> <input class="control__input" name="username" value="miripiruni" id="username-control" /> </div>
```
Comment (miripiruni) [inline docs/en/5-templates-syntax.md]: Spaces

File: docs/en/6-templates-context.md [hunk 18]
```diff
@@ -0,0 +1,276 @@
...
+var html = templates.apply(bemjson);
+```
+'html' contains the string:
+```html
+<div class="b">Hello, username</div>
+```
+
+Using the BEMContext prototype:
+```js
+var bemxjst = require('bem-xjst'); var templates = bemxjst.bemhtml.compile(''); // Extending the context prototype templates.BEMContext.prototype.hi = function(name) { return 'Hello, ' + username; }; // Adding templates templates.compile(function() { block('b').content()(function() { return this.hi('templates'); }); }); var bemjson = { block: 'b' }; // Applying templates var html = templates.apply(bemjson));
```
Comment (miripiruni) [inline docs/en/6-templates-context.md]: Formatting

File: docs/en/7-runtime.md [hunk 19]
```diff
@@ -0,0 +1,159 @@
...
+block('link').tag()(function() {
+    var res = applyNext(); // res === 'a'
+    return res;
+});
+```
+
+
+### applyCtx
+```js
+/** * @param {BEMJSON} bemjson input data * @param {Object} [newctx] * @returns {String} */
```
Comment (miripiruni) [inline docs/en/7-runtime.md]: Formatting

File: docs/en/7-runtime.md [hunk 20]
```diff
@@ -0,0 +1,159 @@
...
+});
+```
+
+```html
+<header class="header"></header> <div class="footer"></div>
+```
+
+### applyNext
+```js
+/** * @param {Object} [newctx] the object whose keys become fields * of the context when executing the template * @returns {*} */
```
Comment (miripiruni) [inline docs/en/7-runtime.md]: Formatting

File: docs/en/7-runtime.md [hunk 21]
```diff
@@ -0,0 +1,159 @@
...
+
+A template with a subpredicate for `*` takes higher priority than a template with a specific name in the subpredicate.. This is related to performance optimization.. In practice, you are unlikely to encounter a situation where this matters.
+
+A subpredicate for a `*` block is true for an empty object.
+
+## <a name="methods"></a>Instructions for controlling runtime
+
+### apply
+```js
+/** * @params {String} modeName mode name * @params {Object} [assignObj] object whose fields will be available * in 'this' in the template body * @returns {*} Returns the result of running the mode */
```
Comment (miripiruni) [inline docs/en/7-runtime.md]: Formatting

File: docs/en/1-about.md [hunk 22]
```diff
@@ -0,0 +1,30 @@
+# About BEM-XJST
+
+## What is BEM-XJST?
+BEM-XJST is a template engine for web development using the BEM methodology.. Starting with version 5.0.0, it contains two engines:
```
Comment (nik-kor) [inline docs/en/1-about.md]: the second point is redandant

**Comment by miripiruni** [comment]
Updated.

- [x] All docs are checked.
- [x] English version are synced with russian (xhtml, js instances for elems, escaping)
- [x] All markdown links are checked.

Ladies and Gentlemen, may I merge it?

**Comment by tadatuta** [comment]
👍 go ahead!

**Pull request merged**